### PR TITLE
chore: auth-wraper docs

### PR DESCRIPTION
**Pull request by pedroharbs**
- Added auth-wrapper docs.

### CODE REVIEW

File: guides/auth-wrapper.md [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+  appId: "demo-sample",
+  getToken: () => Promise.resolve(await DeviceAuth.load(
+    settings
+  ).login(tokenResponse?.refresh_token)),
+});
+​
+```
+​
+You can find a full sample application [here]()
+​
```
Comment (sowmyaraj92) [inline guides/auth-wrapper.md]: Missing link

File: guides/auth-wrapper.md [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+const client = new CogniteClient({
+  project: "my-project",
+  appId: "demo-sample",
+  getToken: () => Promise.resolve(await ImplicitAuth.load(settings).login()),
+});
+​
+```
+​
+You can find a full sample application [here]()
```
Comment (sowmyaraj92) [inline guides/auth-wrapper.md]: Missing link

File: guides/auth-wrapper.md [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+const client = new CogniteClient({
+  project: "my-project",
+  appId: "demo-sample",
+  getToken: () => Promise.resolve(await PkceAuth.load(settings).login()),
+});
+​
+```
+​
+You can find a full sample application [here]()
+​
```
Comment (sowmyaraj92) [inline guides/auth-wrapper.md]: Missing link

File: guides/auth-wrapper.md [hunk 4]
```diff
@@ -0,0 +1,341 @@
...
+  project: "my-project",
+  appId: "demo-sample",
+  getToken: () => Promise.resolve(await PkceAuth.load(settings)
+    .login(tokenResponse?.refresh_token)),
+});
+​
+```
+​
+You can find a full sample application [here]()
+​
```
Comment (sowmyaraj92) [inline guides/auth-wrapper.md]: Missing link

File: guides/auth-wrapper.md [hunk 5]
```diff
@@ -0,0 +1,341 @@
...
+    authority: 'your_authority',
+    client_id: 'your_client_id',
+    grant_type: 'your_grant_type',
+    client_secret: 'your_client_secret',
+    scope: 'your_scope'
+});
+​
+```
+​
+You can find a full sample application [here]()
```
Comment (sowmyaraj92) [inline guides/auth-wrapper.md]: Missing link

File: guides/auth-wrapper.md [hunk 6]
```diff
@@ -0,0 +1,341 @@
...
+    authority: 'your_authority',
+    client_id: 'your_client_id',
+    client_secret: 'your_client_secret',
+    scope: 'your_scope'
+});
+​
+```
+​
+You can find a full sample application [here]()
+​
```
Comment (sowmyaraj92) [inline guides/auth-wrapper.md]: Missing link

File: guides/auth-wrapper.md [hunk 7]
```diff
@@ -0,0 +1,341 @@
...
+    client_id: 'your_client_id',
+    client_id: 'your_client_id',
+    client_secret: 'your_client_secret',
+    scope: 'your_scope'
+});
+​
+```
+​
+You can find a full sample application [here]()
+​
```
Comment (sowmyaraj92) [inline guides/auth-wrapper.md]: Missing link

File: guides/auth-wrapper.md [hunk 8]
```diff
@@ -0,0 +1,341 @@
...
+## API keys
+​
+### getToken metod
+​
+API keys use the same API in the SDK, using `getToken` you will need to set the additional flag `apiKeyMode` to
+true to handle headers appropriately.
```
Comment (sowmyaraj92) [inline guides/auth-wrapper.md]: Unclear information - need more context here.

File: guides/auth-wrapper.md [hunk 9]
```diff
@@ -0,0 +1,341 @@
...
+    scope: 'your_scope'
+});
+​
+```
+​
+You can find a full sample application [here]()
+​
+## API keys
+​
+### getToken metod
```
Comment (sowmyaraj92) [inline guides/auth-wrapper.md]: metod > "method"

File: guides/auth-wrapper.md [hunk 10]
```diff
@@ -0,0 +1,341 @@
...
+  apiKeyMode: true,
+  getToken: () => Promise.resolve('API_KEY_HERE'),
+});
+```
+​
+> **_NOTE:_**  This method will be deprecated soon.
+​
+### credentials metod
+​
+API keys use the same API in the SDK, using `credentials` you will need to set the following object to handle headers appropriately.
```
Comment (sowmyaraj92) [inline guides/auth-wrapper.md]: Same as the above comment.

File: guides/auth-wrapper.md [hunk 11]
```diff
@@ -0,0 +1,337 @@
...
+  - [PKCE Flow w Refresh Token](#oidc-authentication-using-gettoken-w-pkce-flow-w-refresh-token)
+- [OpenID Connect (OIDC) w credentials](#openid-connect-oidc-w-credentials)
+  - [Client Credentials Flow](#oidc-authentication-using-gettoken-w-client-credentials-flow)
+  - [Device Code Flow](#oidc-authentication-using-credentials-w-device-code-flow)
+  - [PKCE Flow](#oidc-authentication-using-credentials-w-pkce-flow)
+- [API keys](#api-keys)
+  - [getToken metod](#gettoken-metod)
+  - [credentials method](#credentials-method)
+- [Manually trigger authentication](#manually-trigger-authentication)
+​
```
Comment (sowmyaraj92) [inline guides/auth-wrapper.md]: Todo:- Will update the entire header in the next review commit!

**Review by sowmyaraj92 — CHANGES_REQUESTED** [review]
@cognite-ornellas There are a few places where we need more context and clarity in the PR, let me know when you're done making the changes.

File: guides/auth-wrapper.md (hunk 1 above)
Comment (cognite-ornellas) [reply guides/auth-wrapper.md]: Is this resolved? Could not find it.

File: guides/auth-wrapper.md (hunk 2 above)
Comment (cognite-ornellas) [reply guides/auth-wrapper.md]: Is this resolved? Could not find it.

File: guides/auth-wrapper.md (hunk 3 above)
Comment (cognite-ornellas) [reply guides/auth-wrapper.md]: Is this resolved? Could not find it.

File: guides/auth-wrapper.md (hunk 4 above)
Comment (cognite-ornellas) [reply guides/auth-wrapper.md]: Is this resolved? Could not find it.

File: guides/auth-wrapper.md (hunk 5 above)
Comment (cognite-ornellas) [reply guides/auth-wrapper.md]: Is this resolved? Could not find it.

File: README.md [hunk 12]
```diff
@@ -24,6 +24,10 @@ There are small bare-bones javascript (and typescript) projects in the `samples/
 They show how to include the cognite SDK in various project setups.
 The samples' [README.md](./samples/README.md) has instructions for running the samples.
 
+## Authentication
+
+We offer two ways to get logged on our sdk, the first and most recommended is using [@cognite/auth-wrapper](./guides/auth-wrapper.md)
+and the second is using [Microsoft MSAL Library](./guides/auth-wrapper.md)
```
Comment (sowmyaraj92) [inline README.md]:
```suggestion
We offer two ways to sign in to our SDK. The first and recommended approach is using [@cognite/auth-wrapper](./guides/auth-wrapper.md)
and the second is using [Microsoft MSAL Library](./guides/auth-wrapper.md)
```

**Review by sowmyaraj92 — APPROVED** [review]
Just one minor change, otherwise we are good to merge! LGTM 🚀

**Review by cognite-ornellas — APPROVED** [review]

**Pull request merged**